estargz/errors.go

```diff
@@ -0,0 +1,46 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package estargz
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Errors provides an abstraction for accumulating multiple errors.
+type Errors []error
+
+// Append adds an error to the Errors
+func (e *Errors) Append(err error) *Errors {
+	if e == nil {
+		return &Errors{err}
+	}
+	errs := append(*e, err)
+	return &errs
+}
+
+// Error implements error
+func (e *Errors) Error() string {
+	points := make([]string, len(*e)+1)
+	points[0] = fmt.Sprintf("%d error(s) occurred:", len(*e))
```

Comment:
should return e[0] directly when length == 1, and return nil when length == 0

Reply:
Good catch, I didn't fix this for nil.  I pushed a change for this.